src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -97,6 +101,7 @@ bool ClusterTaskManager::SchedulePendingTasks() {
       if (node_id_string == self_node_id_.Binary()) {
         // Warning: WaitForTaskArgsRequests must execute (do not let it short
         // circuit if did_schedule is true).
+        scheduling_backpressure_tracker_[sched_cls]++;
```

Comment:
Shouldn't we do this when the task actually gets dispatched? Otherwise, we will not fetch args, etc.

Reply:
hmm if we do that, won't we end up in a case where the task can be scheduled on another node, but is still blocked on this node due to the cap?

Reply:
I think it would make more sense to check the cap on dispatch, then try spillback if the task is not schedulable due to the cap (same as we do if there are no local resources available). That should achieve the same effect and we can reuse an existing codepath to block dispatch.

But in general, I think this corner case is not worth worrying about right now. In the common case, the driver should already have requested leases for the same scheduling class at the other raylet.